client/my-sites/domains/domain-management/edit/domain-types/registered-domain-type.jsx

```diff
@@ -252,6 +255,47 @@ class RegisteredDomainType extends React.Component {
 		);
 	}
 
+	planUpsellForNonPrimaryDomain() {
+		const { domain, hasNonPrimaryDomainsFlag, isDomainOnly, selectedSite, translate } = this.props;
+
+		if (
+			! hasNonPrimaryDomainsFlag ||
+			isDomainOnly ||
+			! domain.pointsToWpcom ||
+			domain.isPrimary ||
+			domain.isWPCOMDomain ||
+			domain.isWpcomStagingDomain
+		) {
+			return null;
+		}
+
+		return (
+			<Banner
```

Comment:
Could you make this a separate component? Two reasons: less code here. But more importantly: `RegisteredDomainType` is the new status page. I understand we're hoping to ship it soon, but just in case, we should add this upsell in `RegisteredDomain` as well. And if it's a separate component, that will be very easy 🙏

Reply:
I was just considering doing that, since pretty much the same component will be reused elsewhere :)